docs/DeveloperGuide.md

```diff
@@ -1,38 +1,323 @@
+Both these classes are important for allowing the User to interact with the application.
+
+<u>Methods</u>
+
+The `Ui` class has the following key methods:
+
+| Method           | Description                   |
+|------------------|-------------------------------|
+| `displayMessage` | Displays a message (`String`) |
+| `getUserInput`   | Reads User Input              |
+
+The Parser class has the following key method:
+
+| Method                 | Description                                               |
+|------------------------|-----------------------------------------------------------|
+| `parseInputToCommand`  | Parses a given user input and returns the related Command |
+
+
+<u>Design Considerations</u>
+
+Low-level I/O operations (eg. stdio) are consolidated in the Ui class such that we can easily switch the I/O methods by 
+modifying only the Ui class. This would make it easier to port the application to other platforms if needed.
+
+Ui class is used as part of exception handling for displaying of error messages to the user for feedback.
+
+The Parser also has some considerations such as
+1. Restricted arguments which should not be used by other developers in their commands. These include
+    1. `/command` -> used for the main command keyword
+    2. `/main` -> used for the main text argument right after the command keyword
+3. Any duplicate arguments will throw an InvalidInputException
+4. All `/` in the argument values should be escaped
+    1. Examples
+       1. `command /argument \/value` -> `argument`: `/value`
+       2. `command /argument value\/value` -> `argument`:`value/value`
+       3. `command /argument value/value` -> `argument`:`value/value` (this is accepted for now, but not recommended)
+       4. `command /argument value\\/value` -> `argument`:`value\/value`
+   2. `/` doesn't need to be escaped for
+       1. commands -> e.g. `/command /argument value1` -> the command is `/command`
+           1. It is discouraged to do so, but the option is left for potential expandability
+       2. arguments -> e.g. `command /argument/param value` -> the argument name is `argument/param`
+    3. Leading and Trailing spaces are ignored, but additional spaces within values (eg. `main  value`) are counted 
+
+### Commands
+
+#### Overview
+
+The abstract Command class has been implemented to introduce an additional layer of abstraction between I/O and command execution, allowing for separation of handling command keywords and executing commands.
+
+
+<u>Implementation Details</u>
+
+The following diagram is an inheritance diagram for Command and its children classes. 
+This has been heavily simplified and only shows the key commands.
+
+![CommandInheritance.png](diagrams%2Fimages%2FCommandInheritance.png)
+
+
+The following diagram is a sequence diagram for execution of Command.
+
+![CommandExecutionSequence.png](diagrams%2Fimages%2FCommandExecutionSequence.png)
+
```

Comment:
The return arrow from isExit() method is labelled as "boolean", which is a variable type. Should it instead be "exit :boolean" or "exit" or something similar, instead of just "boolean"